src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1947 @@
+        hsi_event_scheduled = [
+            ev
+            for ev in self.sim.modules["HealthSystem"].find_events_for_person(person_id)
+            if isinstance(ev[1], (HSI_Wasting_SupplementaryFeedingProgramme_MAM,
+                                  HSI_Wasting_OutpatientTherapeuticProgramme_SAM,
+                                  HSI_Wasting_InpatientTherapeuticCare_ComplicatedSAM))
+        ]
+        if hsi_event_scheduled:
+            return
+
+        # the person may not attend the appt
+        if not self.attendance:
+            return
+
+        available_equipment = []
+        for equip in ['Height Pole (Stadiometer)', 'Weighing scale', 'MUAC tape']:
+            available = rng.random_sample() < HSI_Event.probability_all_equipment_available(self, equip)
+            if available:
+                available_equipment.append(equip)
+        self.add_equipment(set(available_equipment))
+
+        def schedule_tx_by_diagnosis(hsi_event):
+            self.sim.modules['HealthSystem'].schedule_hsi_event(
+                hsi_event=hsi_event(module=self.module, person_id=person_id),
+                priority=0, topen=self.sim.date
+            )
+
+        complications = df.at[person_id, 'un_sam_with_complications']
+
+        # DIAGNOSIS
+        # in future this is the place where misdiagnosing due to equipment unavailability will be happening (but should
+        # be happening also in the generic first appt), however current asserts are made for the case when all equip is
+        # available, hence they will need to be adjusted, see issue #1612
+        # oedema is assumed to be quite obvious if present
+        # in addition, based on performed measurements (depending on what equipment is available)
+
+        # for now, perfect diagnosing is assumed
+        diagnosis = df.at[person_id, 'un_clinical_acute_malnutrition']
+
+        # No intervention if diagnosed as well
+        if diagnosis == 'well':
+            return
+        # SFP intervention if diagnosed as MAM
+        elif diagnosis == 'MAM':
+             schedule_tx_by_diagnosis(HSI_Wasting_SupplementaryFeedingProgramme_MAM)
+        elif (diagnosis == 'SAM') and (not complications):
+            # OTP intervention if diagnosed as uncomplicated SAM
+            schedule_tx_by_diagnosis(HSI_Wasting_OutpatientTherapeuticProgramme_SAM)
+        else:  # (diagnosis == 'SAM') and complications:
+            # ITC intervention if diagnosed as complicated SAM
+            schedule_tx_by_diagnosis(HSI_Wasting_InpatientTherapeuticCare_ComplicatedSAM)
+
+    def did_not_run(self):
+        logger.debug(key="HSI_Wasting_GrowthMonitoring",
+                     data="HSI_Wasting_GrowthMonitoring: did not run"
+                     )
         pass
 
-    def on_birth(self, mother, child):
+
+class HSI_Wasting_SupplementaryFeedingProgramme_MAM(HSI_Event, IndividualScopeEventMixin):
```

Comment:
the two outpatient HSI are very similar -- do they need to be two entirely different classes?